### PR TITLE
dts: x86: quark_se_c1000_devboard: Add device tree support

### PR DESCRIPTION
**Pull request by SavinayDharmappa**
patch add device tree support for develoement board of
quark_se_c1000. Previously pushed patch was flashing binary
at wrong address because of which UART was not working.

1. Added new flash node to the .dtsi file as quark_se_c1000 has two different flash.
2. Added usb_cdc node to the .dtsi file. This is required to support serial over usb.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

### CODE REVIEW

File: dts/x86/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ifeq ($(CONFIG_HAS_DTS),y)
-dtb-$(CONFIG_SOC_QUARK_SE_C1000)	=	arduino_101.dts_compiled
-dtb-$(CONFIG_BOARD_QUARK_D2000_CRB)	=	quark_d2000_crb.dts_compiled
-dtb-$(CONFIG_BOARD_TINYTILE)		=	tinytile.dts_compiled
+dtb-$(CONFIG_SOC_QUARK_SE_C1000)		=	arduino_101.dts_compiled
+dtb-$(CONFIG_BOARD_QUARK_D2000_CRB)		=	quark_d2000_crb.dts_compiled
+dtb-$(CONFIG_BOARD_TINYTILE)			=	tinytile.dts_compiled
+dtb-$(CONFIG_BOARD_QUARK_SE_C1000_DEVBOARD)	=	quark_se_c1000_devboard.dts_compiled
```
Comment (nashif) [inline dts/x86/Makefile]: what is the reason for having all use CONFIG_BOARD excluding arduino 101 which uses CONFIG_SOC?

File: dts/x86/quark_se_c1000_devboard.fixup [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#define CONFIG_UART_QMSI_0_BAUDRATE	INTEL_QMSI_UART_B0002000_CURRENT_SPEED
+#define CONFIG_UART_QMSI_0_NAME		INTEL_QMSI_UART_B0002000_LABEL
```
Comment (nashif) [inline dts/x86/quark_se_c1000_devboard.fixup]: what is this cryptic define using B0002000? where is INTEL_QMSI_UART_B0002000_LABEL defined?

File: dts/x86/quark_se_c1000_devboard.dts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		uart_1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart1;
+		zephyr,bt-uart = &uart0;
+		zephyr,uart-pipe = &uart1;
+		zephyr,bt-mon-uart = &uart1;
```
Comment (nashif) [inline dts/x86/quark_se_c1000_devboard.dts]: what about serial over USB? did you test that with the board, how can I configure the serial console to go over CDC ACM? See related sample for serial usb in samples/subsys/usb

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by jukkar — COMMENTED** [review]
Note that the Jira comment is in wrong format.
It should be
Jira: ZEP-2459

File: dts/x86/Makefile (hunk 1 above)
Comment (galak) [inline dts/x86/Makefile]: Agreed, we should be using BOARD not SOC here.

File: dts/x86/quark_se_c1000_devboard.fixup (hunk 2 above)
Comment (galak) [inline dts/x86/quark_se_c1000_devboard.fixup]: It is generated, the fixup file is meant to map from the generated defines to more "generic" ones that drivers can use.  Eventually the fixup files should go away when we can generate the code/structs directly.

File: dts/x86/mem.h [hunk 4]
```diff
@@ -3,9 +3,15 @@
 
 #define __SIZE_K(x) (x * 1024)
 
-#if defined(CONFIG_SOC_QUARK_SE_C1000)
+#if (defined(CONFIG_SOC_QUARK_SE_C1000) && \
+defined(CONFIG_BOARD_QUARK_SE_C1000_DEVBOARD))
 #define DT_FLASH_SIZE		__SIZE_K(144)
 #define DT_SRAM_SIZE		__SIZE_K(55)
+#define FLASH_ADDRESS           0x40030000
+#elif defined(CONFIG_SOC_QUARK_SE_C1000)
```
Comment (SavinayDharmappa) [inline dts/x86/mem.h]: it is bad implementation i will correct it.

File: dts/x86/quark_se_c1000_devboard.dts (hunk 3 above)
Comment (SavinayDharmappa) [inline dts/x86/quark_se_c1000_devboard.dts]: I tested it. It is not working. I will correct it and push the changes

**Comment by SavinayDharmappa** [comment]
@galak  please review this patch

**Review by nashif — APPROVED** [review]

**Pull request merged**